lib/connection/commands.js

```diff
@@ -484,8 +484,15 @@ Response.prototype.parse = function(options) {
     // Deserialize but keep the array of documents in non-parsed form
     var doc = this.bson.deserialize(document, _options);
 
+    if (doc instanceof Error) {
+      throw doc;
+    }
     // Get the documents
-    this.documents = doc.cursor[documentsReturnedIn];
+    if (doc.cursor) {
+      this.documents = doc.cursor[documentsReturnedIn];
+    } else {
+      throw new Error('Cursor not found');
```

Comment:
This case is usually hit when we get a server error. Can we throw that error instead of a simple `Error`?